cou/apps/base.py

```diff
@@ -68,6 +68,10 @@ class OpenStackApplication(Application):
     packages_to_hold: Optional[list] = field(default=None, init=False)
     wait_timeout: int = field(default=STANDARD_IDLE_TIMEOUT, init=False)
     wait_for_model: bool = field(default=False, init=False)  # waiting only for application itself
+    # Raise error when any unit or app going into "blocked" status
+    raise_on_blocked: bool = field(default=False, init=False)
+    # Raise error when any unit or app going into "error" status
+    raise_on_error: bool = field(default=True, init=True)
```

Comment:
When this is used? I do not see any app, which is changing default value.

Reply:
The vault application will get into error status first then blocked. Without this the COU will raise error when vault get into error status.

Reply:
These values are not configured anywhere, the vault has only
```python
 wait_timeout = LONG_IDLE_TIMEOUT
wait_for_model = True
```
I'm not saying that you should not use them at lines 840 and 841, but no charm is using this class variable, so I suggested removing them.

```python
            coro=self.model.wait_for_active_idle(
                timeout=self.wait_timeout,
                apps=apps,
                raise_on_blocked=False,
                raise_on_error=True,
            ),
```

Reply:
It's used on function `_wait_for_sealed_status`

Reply:
There is no app defined like this:

```python
@AppFactory.register_application(["vault"])
class Vault(AuxiliaryApplication):
    """Application for vault."""

    wait_timeout = LONG_IDLE_TIMEOUT
    wait_for_model = True
    raise_on_blocked = True
    raise_on_error = False
```
So those class variables are never used as class variable. That's why I think using constant would be simpler here.

Reply:
The steps are different.

The general steps are:

`upgrade` -> `wait for sealed` -> `unseal` -> `resolve all` -> `wait for model idle`.

We don't want `raise_on_error` on the last step, it only used on `wait for sealed` step.

Reply:
Ok, now I'm confused. 
> We don't want raise_on_error on the last step, it only used on wait for sealed step.

Check line 841, you are using it there.

I mean the 2 class variables you defined in lines 72 and 74 are used only on lines 840 and 841, nowhere else.
So my suggestion is to remove them and use simple constants
e.g. 
```python
           coro=self.model.wait_for_active_idle(
                timeout=self.wait_timeout,
                apps=apps,
                raise_on_blocked=RAISE_ON_BLOCKED,
                raise_on_error=RAISE_ON_ERROR,
            ),
```
or even better, completely remove them, since those are default values (default behavior) for wait_for_active_idle function.
```python
           coro=self.model.wait_for_active_idle(timeout=self.wait_timeout, apps=apps),
```

Reply:
I see your point.

Reply:
You are adding it here, so it's related.